src/Microsoft.OData.Core/UriParser/Binders/FunctionCallBinder.cs

```diff
@@ -597,13 +633,24 @@ private static ICollection<FunctionParameterToken> HandleComplexOrCollectionPara
         }
 
         /// <summary>
-        /// Determines whether this is a function that we don't bind to a BuiltInFunction
+        /// Determines whether this is a unbound BuiltInFunction according to the configured case-sensitiveness.
         /// </summary>
         /// <param name="functionName">name of the function</param>
-        /// <returns>true if this is a function that we don't bind</returns>
-        private static bool IsUnboundFunction(string functionName)
+        /// <returns>matched unbound function names; null if no matches found.</returns>
+        private string IsUnboundFunction(string functionName)
         {
-            return UnboundFunctionNames.Contains(functionName);
+            foreach (string key in UnboundFunctionNames)
+            {
+                if (key.Equals(functionName,
+                    this.state.Configuration.EnableCaseInsensitiveUriFunctionIdentifier
+                        ? StringComparison.OrdinalIgnoreCase
+                        : StringComparison.Ordinal))
+                {

```

Comment:
since unbound function names are lower case, would be more efficient to cast the value you're looking for to lower case for case insensitive search:

functionName = this.state.Configuration.EnableCaseInsensitiveUriFunctionIdentifier ? functionName.ToLowerInvariant() : functionName
return UnboundFunctionNames.FirstOrDefault(k => k.Equals(functionName, StringComparison.Ordinal));

Reply:
Make sense. Thanks.

---
In reply to: [190097377](https://github.com/OData/odata.net/pull/1156#discussion_r190097377) [](ancestors = 190097377)